odoo/tools/view_validation.py

```diff
@@ -25,6 +30,32 @@ def decorator(fn):
     return decorator
 
 
+def _relaxng(view_type):
+    _relaxng_validator = None
+    with tools.file_open(os.path.join('base', 'rng', '%s_view.rng' % view_type)) as frng:
+        try:
+            relaxng_doc = etree.parse(frng)
+            _relaxng_validator = etree.RelaxNG(relaxng_doc)
+        except Exception:
+            _logger.exception('Failed to load RelaxNG XML schema for views validation')
+    return _relaxng_validator
```

Comment:
Why don't you use the function `_relaxng` to access the cache in `_relaxng_dict`, and hide this implementation detail? And by the way, I would rename the function `relaxng` (no need for privacy here).
```python
_relaxng_cache = {}

def relaxng(view_type):
    """ Return a validator for the given view type, or None. """
    if view_type not in _relaxng_cache:
        with tools.file_open(os.path.join('base', 'rng', '%s_view.rng' % view_type)) as frng:
            try:
                relaxng_doc = etree.parse(frng)
                _relaxng_cache[view_type] = etree.RelaxNG(relaxng_doc)
            except Exception:
                _logger.exception('Failed to load RelaxNG XML schema for views validation')
                _relaxng_cache[view_type] = None
    return _relaxng_cache[view_type]
```